### PR TITLE
Feature engineering module

### PR DESCRIPTION
**Pull request by bradleyboehmke**
This PR incorporates the Feature Engineering module that will be taught during day 2 of the workshop.

### CODE REVIEW

**Comment by eswan18** [comment]
One small thing on something I think you merged earlier: you use the syntax `_ = adult_census.hist(figsize=(20, 14))`, but in the intro we teach students to use a semicolon for the same purpose (avoiding the ugly `<matplotlib.axes object ...>`): `adult_census.hist(figsize=(20, 14));`. For consistency, it would be nice to keep with that syntax here.

File: notebooks/05-feat_eng.ipynb [hunk 1]
```diff
@@ -0,0 +1,2281 @@
...
+    "slideshow": {
+     "slide_type": "skip"
+    },
+    "tags": []
+   },
+   "source": [
+    " \n",
+    "<div class=\"admonition tip alert alert-warning\">\n",
+    "    <p class=\"first admonition-title\" style=\"font-weight: bold;\"><b>Tip</b></p>\n",
+    "    <p class=\"last\">We can use the <tt class=\"docutils literal\">handle_unknown</tt> parameter to ignore categories rare categories that may show up in test data but were not present in the training data.</p>\n",
```
Comment (eswan18) [inline notebooks/05-feat_eng.ipynb]: typo: "categories rare categories"

File: notebooks/05-feat_eng.ipynb [hunk 2]
```diff
@@ -0,0 +1,2281 @@
...
+    "`ColumnTransformer`:\n",
+    "\n",
+    "![columntransformer diagram](images/api_diagram-columntransformer.svg)\n",
+    "\n",
+    "A `ColumnTransformer` does the following:\n",
+    "\n",
+    "* It **splits the columns** of the original dataset based on the column names\n",
+    "  or indices provided. We will obtain as many subsets as the number of\n",
+    "  transformers passed into the `ColumnTransformer`.\n",
+    "* It **transforms each subsets**. A specific transformer is applied to\n",
```
Comment (eswan18) [inline notebooks/05-feat_eng.ipynb]: typo: "each subsets"

File: notebooks/05-feat_eng.ipynb [hunk 3]
```diff
@@ -0,0 +1,2281 @@
...
+    "slideshow": {
+     "slide_type": "skip"
+    },
+    "tags": []
+   },
+   "source": [
+    "<div class=\"admonition warning alert alert-danger\">\n",
+    "    <p class=\"first admonition-title\" style=\"font-weight: bold;\"><b>Warning</b></p>\n",
+    "<p class=\"last\">Including non-scaled data can cause some algorithms to iterate\n",
+    "longer in order to converge. Since our categorical features are not scaled its often recommended to increase the number of allowed iterations for linear models.</p>\n",
```
Comment (eswan18) [inline notebooks/05-feat_eng.ipynb]: typo: "its" should be "it's"

**Review by eswan18 — CHANGES_REQUESTED** [review]
Just a couple tiny typos

File: notebooks/05-feat_eng.ipynb (hunk 1 above)
Comment (bradleyboehmke) [reply notebooks/05-feat_eng.ipynb]: Good catch - addressed in #27

File: notebooks/05-feat_eng.ipynb (hunk 2 above)
Comment (bradleyboehmke) [reply notebooks/05-feat_eng.ipynb]: Good catch - addressed in #27

File: notebooks/05-feat_eng.ipynb (hunk 3 above)
Comment (bradleyboehmke) [reply notebooks/05-feat_eng.ipynb]: Good catch - addressed in #27

**Review by eswan18 — APPROVED** [review]

**Pull request merged**